src/main/java/edu/ucsb/cs156/example/services/UCSBCurriculumService.java

```diff
@@ -0,0 +1,99 @@
+package edu.ucsb.courses.services;
+
+import java.util.Arrays;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Service object that wraps the UCSB Academic Curriculum API
+ */
+@Service
+public class UCSBCurriculumService  {
+
+    private Logger logger = LoggerFactory.getLogger(UCSBCurriculumService.class);
+
+    @Value("${app.ucsb.api.consumer_key}")
+    private String apiKey;
+
+    private RestTemplate restTemplate = new RestTemplate();
+
+    public String getJSON(String subjectArea, String quarter, String courseLevel) {
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("ucsb-api-version", "1.0");
+        headers.set("ucsb-api-key", this.apiKey);
+
+        HttpEntity<String> entity = new HttpEntity<>("body", headers);
+
+        String uri = "https://api.ucsb.edu/academics/curriculums/v1/classes/search";
+        String params = String.format(
```

Comment:
Using `String#format` to construct URLs may lead to subtle bugs when special characters are substituted into the URL. There are a few ways to rewrite this code to avoid this issue, including `UriComponentsBuilder` and the [`uriVariables` parameter used in team03](https://github.com/ucsb-cs156-w22/STARTER-team03/blob/main/src/main/java/edu/ucsb/cs156/example/services/RedditPostService.java).

Reply:
Agreed.  Please see the way this is done in team03 at the link Bryan provided, and change the code to avoid using String.format.

Even though this was done in the S21 code, we have higher standards now. :-)